caf-audit/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditIndexManager.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2015-2017 Hewlett Packard Enterprise Development LP.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.auditing.elastic;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.ResourceAlreadyExistsException;
+import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+
+public class ElasticAuditIndexManager {
+
+    private static final Logger LOG = LogManager.getLogger(ElasticAuditIndexManager.class.getName());
+
+    private final TransportClient transportClient;
+    private final LoadingCache<String, String> indexCache;
+    private final ElasticAuditConfiguration config;
+
+    public ElasticAuditIndexManager(ElasticAuditConfiguration config, TransportClient transportClient) {
+        this.transportClient = transportClient;
+        this.config = config;
+
+        //  Configure in memory cache to hold list of Elasticsearch indexes already created.
+        indexCache = CacheBuilder
+                .newBuilder()
+                .build(
+                        //  CacheLoader used to create new index and load into cache if not already cached.
+                        new CacheLoader<String, String>() {
+                            public String load(String indexName) {
+                                createIndex(indexName);
+                                return indexName;
+                            }
+                        }
+                );
+    }
+
+    /**
+     *  Method used to query the index cache.
+     */
+    public String getIndex(String indexName){
+        return indexCache.getUnchecked(indexName);
+    }
+
+    /**
+     *  Returns immediately if index already exists in Elasticsearch, otherwise it creates a new index in Elasticsearch.
+     */
+    private void createIndex(String indexName){
+        final String indexAlreadyExistsMessage = "Index " + indexName + " already exists";
+
+        //  Return immediately if index already exists. Otherwise, create a new one.
+        if (indexExists(indexName)) {
+            LOG.debug(indexAlreadyExistsMessage);
+            return;
```

Comment:
This is totally down to personal preference so I don't think should be picked up in code reviews.

https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement